docs/knowledge_base/tips/colab_with_ngrok/index.md

```diff
@@ -0,0 +1,184 @@
+![Sharing our Taipy application in Google Colab with Ngrok](Sharing_Taipy_Ngrok_5_1.png){width=70%}
+
+Now, please click on the Ngrok Public URL link to access our application.
+
+![Sharing our Taipy application in Google Colab with Ngrok](taipy_ngrok_app.gif){width=100%}
+
+Congratulations! You have successfully created and launched our multi-page Taipy application 
+from Colab!!
+
+On the first page, you can enter a word or sentence and see its scores determined by an NLP 
+algorithm. The results are displayed in a table and a graph. On the second page, you can upload 
+a file and see the analysis of an entire text.
+
+To learn more, visit our 
+[Getting Started](https://docs.taipy.io/en/develop/getting_started/) page. You can also share 
+this application with anyone on the internet.
+
+## Update the delay parameter
+
+Our application, accessed through Ngrok, is updated at intervals determined by a delay parameter.
+When we input a word or sentence, you may notice that it's not always registered correctly 
+because the page refreshes several times. This happens because of the time it takes for data to 
+travel over the internet, causing delays.
+
+To address this issue, you can modify the `change_delay` parameter in one of the following ways:
+
+- **Locally**: You can make changes to this parameter locally, directly within the Taipy visual 
+  element.
+  ![Update the delay parameter](Sharing_Taipy_Ngrok_6.png){width=100%}
+
+- **Globally**: To adjust the delay for all of Taipy's visual elements.
+  ![Update the delay parameter](Sharing_Taipy_Ngrok_7.png){width=100%}
+
+## Reloading after modification
+
+As it stands, when you make changes to a cell, there's no need to restart the kernel. You can 
+simply rerun the Notebook to see your changes in action. However, in Ngrok's free version, you 
+are restricted to three re-executions, which might require a kernel restart.
+
+Nevertheless, Taipy offers built-in functions that enhance the notebook experience with Taipy. 
+This eliminates the limitations of the Ngrok free version 
+and enables easy updates with fewer re-executions required. You can learn more about this in the 
+[linked article](https://www.taipy.io/tips/taipy-gui/taipy-in-jupyter-notebooks/).
+
+## Modification of the markdown with
+
+`Page.set_content` here are the new cells to add:
+
+1. Import Markdown:
+   ![Modification of the markdown](Sharing_Taipy_Ngrok_8.png){width=100%}
+
+2. Set empty new page:
+   ![Modification of the markdown](Sharing_Taipy_Ngrok_9.png){width=100%}
+
+3. Set content to `new_page`:
+   ![Modification of the markdown](Sharing_Taipy_Ngrok_10.png){width=100%}
+
+4. Update the `pages` definition:
+   ![Modification of the markdown](Sharing_Taipy_Ngrok_11.png){width=100%}
+
+## Variable modification with `gui.reload`
+
+1. Add this step
+   ![Variable modification](Sharing_Taipy_Ngrok_12.png){width=100%}
+
+2. Update your `Gui.run`
```

Comment:
I believe we should preferably document `tp.run(gui)` instead of gui.run().
This also concern the picture.

Reply:
OK I will update it, thanks.